packages/extractor/src/extractors/SpotifyExtractor.ts

```diff
@@ -50,10 +50,53 @@ export class SpotifyExtractor extends BridgedExtractor<SpotifyExtractorInit> {
     }
 
     public async getRelatedTracks(track: Track) {
-        return await this.handle(track.author || track.title, {
-            type: QueryType.SPOTIFY_SEARCH,
-            requestedBy: track.requestedBy
+        const useSpotifyApi = true;
+
+        // if useSpotifyApi is false, we will use old way to retrieve related tracks
+        if (!useSpotifyApi) {
+            return await this.handle(track.author || track.title, {
+                type: QueryType.SPOTIFY_SEARCH,
+                requestedBy: track.requestedBy
+            });
+        }
+
+        const trackMetadata = track.metadata as { source: SpotifySong };
+        const trackId = trackMetadata.source.id;
+        const artistId = trackMetadata.source.artists[0].uri.split(':')[2];
+        if (!trackId || !artistId) return this.createResponse();
+
+        const data = await this.internal.getRelatedTracks(trackId, artistId, 5);
```

Comment:
We dont need to fetch 5 tracks here, autoplay only needs 1 track.

Reply:
Yeah. Just used this to easily see quality of recommendations from Spotify API. Final implementation will have 1 as limit hardcoded. This parameter will be removed.